protocol/x/vault/types/msg_deposit_to_vault_test.go

```diff
@@ -0,0 +1,64 @@
+package types_test
+
+import (
+	"testing"
+
+	"github.com/dydxprotocol/v4-chain/protocol/dtypes"
+	"github.com/dydxprotocol/v4-chain/protocol/testutil/constants"
+	satypes "github.com/dydxprotocol/v4-chain/protocol/x/subaccounts/types"
+	"github.com/dydxprotocol/v4-chain/protocol/x/vault/types"
+	"github.com/stretchr/testify/require"
+)
+
+func TestMsgDepositToVault_ValidateBasic(t *testing.T) {
+	tests := map[string]struct {
+		msg         types.MsgDepositToVault
+		expectedErr string
+	}{
+		"Success": {
+			msg: types.MsgDepositToVault{
+				VaultId:       &constants.Vault_Clob_0,
+				SubaccountId:  &constants.Alice_Num0,
+				QuoteQuantums: dtypes.NewInt(1),
+			},
+		},
+		"Failure: zero quote quantums": {
+			msg: types.MsgDepositToVault{
+				VaultId:       &constants.Vault_Clob_0,
+				SubaccountId:  &constants.Alice_Num0,
+				QuoteQuantums: dtypes.NewInt(0),
+			},
+			expectedErr: "Deposit amount is invalid",
+		},
+		"Failure: negative quote quantums": {
+			msg: types.MsgDepositToVault{
+				VaultId:       &constants.Vault_Clob_0,
+				SubaccountId:  &constants.Alice_Num0,
+				QuoteQuantums: dtypes.NewInt(-1),
+			},
+			expectedErr: "Deposit amount is invalid",
+		},
+		"Failure: invalid subaccount owner": {
+			msg: types.MsgDepositToVault{
+				VaultId: &constants.Vault_Clob_0,
+				SubaccountId: &satypes.SubaccountId{
+					Owner:  "invalid-owner",
+					Number: 0,
+				},
+				QuoteQuantums: dtypes.NewInt(-1),
```

Comment:
nit: should you make QuoteQuantums a positive value here as to isolate which error we are going for?

Reply:
good catch. missed it